master/internal/resourcemanagers/agent/agent.go

```diff
@@ -123,9 +139,14 @@ func (a *agent) receive(ctx *actor.Context, msg interface{}) error {
 		}
 
 		var masterSetAgentOptions aproto.AgentMessage
-		if a.awaitingReconnect && a.agentReattachEnabled {
+		fmt.Println("websocketconnected", a.awaitingReconnect, a.agentReattachEnabled)
+		// For reconnecting containers, or when reattach is enabled, try to synchronize containers.
+		// Flush them otherwise.
+		reconnect, _ := msg.IsReconnect()
+		if a.awaitingReconnect && (a.agentReattachEnabled || reconnect) {
```

Comment:
i dont get why this is 'or' and not just all ands

Reply:
intent was to have container revalidation work:
 - when reattach is on or off, on valid reconnects
 - when reattach is on, also do it on initial connect (because restored agent actor also has `awaitingReconnect = true`

Reply:
maybe enumerate all these different cases in a comment